temba/api/v2/internals/channels/views.py

```diff
@@ -45,6 +52,42 @@ def post(self, request: Request):
         return Response(response)
 
 
+class InternalChannelView(APIViewMixin, APIView):
+    authentication_classes = [InternalOIDCAuthentication]
+    permission_classes = [IsAuthenticated, CanCommunicateInternally]
+
+    def get(self, request: Request):
+        params = request.query_params
+        project_uuid = params.get("project_uuid")
+
+        if project_uuid is None:
+            return Response(status=400, data={"error": "project_uuid is required"})
+
+        try:
+            org = Org.objects.get(proj_uuid=project_uuid)
+        except Org.DoesNotExist:
+            return Response(status=404, data={"error": "Project not found"})
+
+        response = {"results": []}
+
+        channels = Channel.objects.filter(org=org, is_active=True)
+        for channel in channels:
+            channel_data = {
+                "uuid": str(channel.uuid),
+                "channel_type": channel.channel_type,
+                "name": channel.name,
+            }
+            if channel.channel_type == "WAC":
```

Comment:
Dont think that this is needed, we can use the default config keys, but is up to you, keep it if you like it